### PR TITLE
CMR-10210: Fix 502 gateway error on cmr-stac index.html/documentation & CMR-10208: Add content type in stac response to match service-desc

### PR DESCRIPTION
**Pull request by eudoroolivares2016**
# Overview

### What is the feature?

Regenerating the `index.html `and including some documentation on using this and some fixes to some markdown linting
also fixes the mimetype discrepancy between the openapi documentation endpoint and what we had listed. The content-type in the response is `text/yaml` but, that is not a valid mimetype on `https://www.iana.org/assignments/media-types/media-types.xhtml` the value `application/yaml` is applicable for the response type.
### What is the Solution?

Used the `redocly` tool to regenerate the `index.html` files update the mimetype for the openapi documentation endpoint

### What areas of the application does this impact?

The index documentation file and just documentation itself

# Testing

### Reproduction steps

- **Environment for testing:*SIT*
- **Collection to test with:*NA*

1. Ensure that documentation is no longer getting a 502 error on `https://cmr.sit.earthdata.nasa.gov/stac/docs/index.html`
2. Ensure that the `service-desc` in the root catalog (https://cmr.sit.earthdata.nasa.gov/stac/) and for a providers catalog https://cmr.sit.earthdata.nasa.gov/stac/<any-provider> response has the correct mimetype (`application/yaml`)

### Attachments

image

# Checklist

- [ ] I have added automated tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings

### CODE REVIEW

File: src/routes/rootCatalog.ts [hunk 1]
```diff
@@ -25,7 +25,7 @@ const selfLinks = (req: Request): Link[] => {
     },
     {
       rel: "service-desc",
-      href: `${stacRoot}/resources/swagger.json`,
+      href: `https://api.stacspec.org/v1.0.0-beta.1/openapi.yaml`,
       title: "OpenAI Documentation",
       type: "application/vnd.oai.openapi+json;version=3.0",
```
Comment (william-valencia) [inline src/routes/rootCatalog.ts:30]: Should we update the type here since its a yaml file and not json anymore?

Comment (eudoroolivares2016) [reply src/routes/rootCatalog.ts]: I'd forgotten about this ticket https://bugs.earthdata.nasa.gov/browse/CMR-10208 I'll try to make sure I'm doing that here

File: src/routes/rootCatalog.ts [hunk 2]
```diff
@@ -25,9 +25,9 @@ const selfLinks = (req: Request): Link[] => {
     },
     {
       rel: "service-desc",
-      href: `${stacRoot}/resources/swagger.json`,
+      href: `https://api.stacspec.org/v1.0.0-beta.1/openapi.yaml`,
       title: "OpenAI Documentation",
```
Comment (ircwaves) [inline src/routes/rootCatalog.ts:29]: Should this be:
```suggestion
      title: "OpenAPI Documentation",
```

File: src/routes/rootCatalog.ts (hunk 1 above)
Comment (eudoroolivares2016) [reply src/routes/rootCatalog.ts]: The GET request to ` https://api.stacspec.org/v1.0.0-beta.1/openapi.yaml` has a response with `text/yaml` as the `Content-Type`

File: src/routes/rootCatalog.ts (hunk 2 above)
Comment (eudoroolivares2016) [reply src/routes/rootCatalog.ts:29]: Great catch so used to seeing openAI these days didn't look weird

File: src/routes/rootCatalog.ts (hunk 1 above)
Comment (william-valencia) [reply src/routes/rootCatalog.ts]: Take a look at this: https://stackoverflow.com/questions/52541842/what-is-the-media-type-of-an-openapi-schema

Still not sure, what it should exactly be though...

Comment (william-valencia) [reply src/routes/rootCatalog.ts]: https://httptoolkit.com/blog/yaml-media-type-rfc/

Comment (william-valencia) [reply src/routes/rootCatalog.ts]: According to IANA the only official media types that I think would be applicable for this yaml file would be:
application/yaml or application/vnd.oai.workflows+yaml
from
https://www.iana.org/assignments/media-types/media-types.xhtml

Comment (william-valencia) [reply src/routes/rootCatalog.ts]: https://datatracker.ietf.org/doc/draft-ietf-httpapi-rest-api-mediatypes/ Looks like this is the latest draft and it is proposing this one:
application/openapi+yaml

**Review by dmistry1 — APPROVED** [review]

**Review by william-valencia — APPROVED** [review]

**Pull request merged**